tutorials/scripting/c_sharp/index.rst

```diff
@@ -1,17 +1,54 @@
-C#
-===
+C#/.NET
+=======
+
+C# is a high-level programming language developed by Microsoft. Godot supports
+C# as an option for a scripting language, alongside Godot's own :ref:`GDScript<toc-learn-scripting-gdscript>`.
+
+The standard Godot executable does not contain C# support out of the box. Instead,
+to enable C# support for your project you need to `download a .NET version <https://godotengine.org/download/>`_
+of the editor from the Godot website.
+
 
 .. toctree::
    :maxdepth: 1
    :name: toc-learn-scripting-C#
 
    c_sharp_basics
    c_sharp_features
+   c_sharp_style_guide
+   diagnostics/index
+
+Godot API for C#
+----------------
+
+As a general purpose game engine Godot offers some high-level features as a part
+of its API. Articles below explain how these features intergrate into C# and how
+C# API may be different from GDScript.
+
+.. toctree::
+   :maxdepth: 1
+   :name: toc-learn-scripting-C#-differences
+
    c_sharp_differences
    c_sharp_collections
    c_sharp_variant
    c_sharp_signals
    c_sharp_exports
    c_sharp_global_classes
-   c_sharp_style_guide
-   diagnostics/index
+
+.. _doc_c_sharp_platforms:
+
+C# platform support
```

Comment:
I love the new index page, but I'm not sure if this is the best place for this section. We have similar sections in `c_sharp_basics` (_Current gotchas and known issues_ or _Common pitfalls_) but I'm not sure if that's better, to be honest I think the `c_sharp_basics` page should probably be split into multiple pages.

Reply:
I know what you mean, and I was thinking removing some stuff from the basics page that can be now found on the index page... but ultimately decided that a bit of duplication is fine. These are likely to be the two landing pages, so it makes sense to have a bit of important information on both of them.

I do support reorganizing the docs, though. I'm not entirely sure why we have articles framed as introduction to C#. That's probably unnecessary and we could go straight into Godot-specific stuff (with some notes here and there that link Godot API to C# concepts).

I could give it a go as time allows, but it would be harder for me to find a good angle here since I'm neither a C# user nor maintainer 🙃 But I can help, if you're interested.

Reply:
I would love that, maybe we can do that after the 4.2 release. For now I think this is good.